project-koku/koku
# Squash reporting migrations

Fixes [COST-2052](https://issues.redhat.com/browse/COST-2052)

Changes proposed in this PR:
* Squash the reporting migrations by recreating them

#### Testing Instructions:
1. The gist here is that this migration will not appear to apply anything during `make run-migrations`, and running `make make-migrations` should not create any new migrations.
2. Look in the DB and see:
```
postgres=# select * from django_migrations;
 id |       app        |                               name                               |            applied            
----+------------------+------------------------------------------------------------------+-------------------------------
  1 | api              | 0001_initial                                                     | 2022-03-03 19:28:44.653626+00
....
 82 | reporting        | 0250_squash                                                      | 2022-03-03 19:29:26.498551+00
```


#### Test creating a new migration:
1. Apply migrations from this branch.
3. create a test model, such as:
```
class TestModel(models.Model):
    uuid = models.UUIDField(primary_key=True)
```
4. run `make make-migrations`
5. Examine the new migration file. See that its dependency refers to `0250_squash` like:
```
dependencies = [("reporting", "0250_squash")]
```

#### Testing with a fresh DB:
1. Apply the migrations from the branch.
2. See that it _only_ runs the `0250_squash` reporting migration:
```
[standard:public]   Applying cost_models.0004_auto_20210903_1559...
[standard:public]  OK
[standard:public]   Applying reporting.0250_squash...
[standard:public]  OK
[standard:public]   Applying reporting_common.0001_initial...
[standard:public]  OK
```
3. run `make create-test-customer`
4. Again, see that only the `0250_squash` runs:
```
koku_server       |   Applying cost_models.0004_auto_20210903_1559...
koku_server       |  OK
koku_server       |   Applying reporting.0250_squash...
koku_server       | [2022-03-03 19:30:25,251] INFO None Creating TABLE reporting_awsaccountalias
...
koku_server       | [2022-03-03 19:30:38,339] INFO None Creating TABLE partitioned_tables
koku_server       | [2022-03-03 19:30:38,382] INFO None Creating TABLE user_settings
koku_server       | [2022-03-03 19:30:38,553] INFO None ALTERING IDS
koku_server       | [2022-03-03 19:30:38,602] INFO None CREATING INDICES
koku_server       |  OK
koku_server       |   Applying reporting_common.0001_initial...
```
